### ISSUE TITLE
Finetuned RFDETR-small model outputs detection with high IOU

### ISSUE THREAD

**Issue by agvdndor** [roboflow/rf-detr#434]
Hi! We have finetuned a RFDETR-small model for object detection of a single custom class. In most cases it works very well, but there's an environment in which the model frequently outputs two detections for a single ground truth object in the image that have a very high IOU (>0.9). Our use case involves object counting so this is a breaking issue.

Specifically, We have trained a RFDETR-small model (with `rfdetr=1.3.0`) exported the model to onnx (opset 16) and are doing inference via a [Triton inference server (v2.49)](https://github.com/triton-inference-server/server/releases/tag/v2.49.0) with `onnxruntime=1.18`.

The obvious answer is to apply NMS postprocessing to resolve this, but since not requiring NMS (and the inference speed gain is brings) is one of the selling points of using RFDETR over YOLO I was wondering to what degree this is expected behavior or not! The expected number of detections per image for our use case is limited (<20) so the the NMS overhead would be minimal as well.

**Comment by agvdndor** [comment]
Aha interesting. We indeed only trained it for 20 epochs because the dataset is not that challenging and we're already having a near-perfect score after 20 epochs. I'll try training for longer and see if the behavior still occurs. 

Can I assume that once the loss curves converge we are done training and this should no longer happen? 

(Opset 16 was used because it has to run on an older device as well where we currently cannot upgrade the triton server to a more recent version)

**Comment by isaacrob-roboflow** [comment]
Loss will be minimized when this behavior doesn't occur, so ideally once it converges it is in that minimal state such that it won't. But hard to say a priori of course

**Comment by isaacrob-roboflow** [comment]
These models were all discovered with NAS and finetuned (see the newly released paper for details) so other resolutions were very much within domain until the final stage. I don't have specific numbers but it's probably safe, just not Pareto optimal on COCO. But we show that running NAS directly on other datasets is better than transferring optimal COCO architectures so this may be one of those cases. 

Are the boxes exactly the same but with different labels? Or do they just have very high overlap but not exactly the same

**Comment by Borda** [comment]
Hey — checking in on this one. Is this still an issue on the latest version (v1.6.1)? If so, let us know and we'll keep investigating. If not, we'll close this out to keep the tracker tidy. Feel free to reopen anytime if it resurfaces!

**Comment by agvdndor** [comment]
Thanks for the follow-up @Borda. I'll bump the version, retrain and try again. Have the model weights/training regime been updated since `v1.3.0`?

**Comment by Borda** [comment]
Good question, and I don't since I joined with 1.4 and it implemented md5 to track it...
@isaacrob would know better 🦝

**Comment by isaacrob** [comment]
There have been some changes but all in the name of improving efficiency while keeping behavior the same
